### ISSUE TITLE
Feature Request: memory_ratio ipolicy parameter for overcommitting memory

### ISSUE THREAD

**Issue by Ganeti-Issues-Migrator** [ganeti/ganeti#1178]
Originally reported of Google Code with ID 1122.

```
Ganeti’s current memory resource model assumes that each instance reserves a fixed amount of memory. KVM instances, however, from the point of view of the host operating system, do not initially use all of their memory. Instead, a KVM process’s memory usage will gradually approach its cap. Once the instances stabilize at their maximum ram usage, Ganeti is able to properly track memory usage. However, because Ganeti tracks memory free on a node instead of memory allocated, Ganeti’s current memory accounting cannot take into account memory overcommitting.

For KVM memory overcommitting, the steady state of the node’s memory usage is (assuming no cpu or disk constraints) completely full memory and swap, with most of the swapped-out memory being free memory in the instances. In this state, Ganeti will consider the node to have no free memory and will refuse to place more instances on the node, even if the desired overcommit ratio has not been reached.

To resolve this, Ganeti’s memory model should be modified to match its cpu model: a mem_ratio ipolicy parameter (similar to vcpu_ratio) should be added and Ganeti should track instance max memory (as it does with instance vcpus) instead of free memory on the node.The semantics of these parameters should match those of their cpu counterparts.
```
Originally added on 2015-07-23 21:50:40 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
I agree that this is a valuable feature request. However, the change
is so fundamental to the various parts of Ganeti, that a detailed
design will be necessary. It probably won't happen in the near future.
However I'll see what I can do to push this idea forward.
```
Originally added on 2015-07-24 08:18:47 +0000 UTC.

Changed State: Accepted
Added to Milestone: Unplanned

**Comment by Ganeti-Issues-Migrator** [comment]
```
This is implemented now in stable-2.17. For the design see
http://docs.ganeti.org/ganeti/2.17/html/design-memory-over-commitment.html
```
Originally added on 2015-10-23 10:03:50 +0000 UTC.

Changed State: Fixed
Added to Milestone: Release2.17
Removed from Milestone: Unplanned
